# Taller 2

* Tomás Otéro
* Adriana Velásquez

In [22]:
import numpy as np
import matplotlib.pyplot as plt
import sympy as sym
from sympy import lambdify

### 1. Derivación: Problema 8 

Es posible construir una aproximación de orden O(h^2) para la derivada progresiva. Para
tal propósito, se escribe el polinomio de interpolación de grado 2 para el conjunto soporte Ω = {(x0, f(x0)),(x1, f(x1)),(x2, f(x2))}, y posteriormente se calcula la derivada de este
polinomio.

### 2. Raíces de Polinomios: Problema 5

(SymPy) Calcular todas las ra´ıces reales de los primeros 20 polinomios de Laguerre.

In [104]:


#Definir el polinomio n de laguerre
def laguerre_Pol(n):
    x = sym.Symbol('x',real=True, positive=True)
    Der = sym.exp(-x)*x**n
    L = (sym.exp(x)/sym.factorial(n))*sym.diff(Der,x,n,evaluate=True)
    return L

def Derivative(f,x,h=1e-4):
    return (f(x+h)-f(x-h))/(2*h)


#Definir las raices del polinomio n de laguerre

def GetNewtonRaphson(f,df,xn,itmax = 100, precision=1e-5):
    
    error = 1
    it = 0
    
    while error > precision and it <= itmax:
        
        try:
            
            xn1 = xn - f(xn)/df(f,xn)
            
            error = np.abs(f(xn)/df(f,xn))
            
            #print(error)
            
        except ZeroDivisionError:
            
            print('Division por cero')
            
        it += 1
        xn = xn1
    
    if it == itmax:
        False
    else:
        return xn
    
    
def GetAllRoots(n, x,tolerancia=8):
    
    Roots = np.array([])
    x_sym = sym.Symbol('x', real=True, positive=True)
    Function = lambdify(x_sym, laguerre_Pol(n))
    
    for i in x:
        
        root = GetNewtonRaphson(Function,Derivative,i)
        
        if root != False:
            
            croot = np.round(root, tolerancia)
            
            if croot not in Roots:
                Roots = np.append(Roots,croot)
                
    Roots.sort()
        
    return Roots

x = np.linspace(-1,4,1000)

print('Polinomio 1')
print('1' )
print('Sus raices son: No tiene raices' )
print('  ')

for i in range(29):
    print('Polinomio ', i+1 )
    #print('$', sym.latex(laguerre_Pol(i+1)), '$')
    print('Sus raices son: ')
    print((GetAllRoots(i+1, x)))
    print('  ')

Polinomio 1
1
Sus raices son: No tiene raices
  
Polinomio  1
Sus raices son: 
[1.]
  
Polinomio  2
Sus raices son: 
[0.58578644 3.41421356]
  
Polinomio  3
Sus raices son: 
[0.41577456 2.29428036 6.28994508]
  
Polinomio  4
Sus raices son: 
[0.32254769 1.7457611  4.5366203  9.39507091]
  
Polinomio  5
Sus raices son: 
[ 0.26356032  1.41340306  3.59642577  7.08581001 12.64080084]
  
Polinomio  6
Sus raices son: 
[ 0.2228466   1.1889321   2.99273633  5.77514357  9.83746742 15.98287398]
  
Polinomio  7
Sus raices son: 
[ 0.19304368  1.0266649   2.56787674  4.90035308  8.18215344 12.73418029
 19.39572786]
  
Polinomio  8
Sus raices son: 
[ 0.17027963  0.90370178  2.25108663  4.26670017  7.0459054  10.75851601
 15.74067864 22.86313174]
  
Polinomio  9
Sus raices son: 
[ 0.15232223  0.80722002  2.00513516  3.78347397  6.20495678  9.37298525
 13.46623691 18.83359779 26.37407189]
  
Polinomio  10
Sus raices son: 
[ 0.13779347  0.72945455  1.8083429   3.4014337   5.55249614  8.33015275
 11.843

### Polinomios y raices de laguerre

A continuación se muestran los primeros 20 polinomios de Laguerre mostrados en latex a partir del codigo de arriba.

El polinomio 1 es: 
1
No tiene raices
  
El polinomio  1 es: 
$ \left(- x e^{- x} + e^{- x}\right) e^{x} $
Sus raices son: 
$ \mathtt{\text{[1.]}} $
  
El polinomio  2 es: 
$ \frac{x^{2}}{2} - 2 x + 1 $
Sus raices son: 
$ \mathtt{\text{[0.58578644 3.41421356]}} $
  
El polinomio  3 es: 
$ - \frac{x^{3}}{6} + \frac{3 x^{2}}{2} - 3 x + 1 $
Sus raices son: 
$ \mathtt{\text{[0.41577456 2.29428036 6.28994508]}} $
  
El polinomio  4 es: 
$ \frac{x^{4}}{24} - \frac{2 x^{3}}{3} + 3 x^{2} - 4 x + 1 $
Sus raices son: 
$ \mathtt{\text{[0.32254769 1.7457611  4.5366203  9.39507091]}} $
  
El polinomio  5 es: 
$ - \frac{x^{5}}{120} + \frac{5 x^{4}}{24} - \frac{5 x^{3}}{3} + 5 x^{2} - 5 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.26356032  1.41340306  3.59642577  7.08581001 12.64080084]}} $
  
El polinomio  6 es: 
$ \frac{x^{6}}{720} - \frac{x^{5}}{20} + \frac{5 x^{4}}{8} - \frac{10 x^{3}}{3} + \frac{15 x^{2}}{2} - 6 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.2228466   1.1889321   2.99273633  5.77514357  9.83746742 15.98287398]}} $
  
El polinomio  7 es: 
$ - \frac{x^{7}}{5040} + \frac{7 x^{6}}{720} - \frac{7 x^{5}}{40} + \frac{35 x^{4}}{24} - \frac{35 x^{3}}{6} + \frac{21 x^{2}}{2} - 7 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.19304368  1.0266649   2.56787674  4.90035308  8.18215344 12.73418029
 19.39572786]}} $
  
El polinomio  8 es: 
$ \frac{x^{8}}{40320} - \frac{x^{7}}{630} + \frac{7 x^{6}}{180} - \frac{7 x^{5}}{15} + \frac{35 x^{4}}{12} - \frac{28 x^{3}}{3} + 14 x^{2} - 8 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.17027963  0.90370178  2.25108663  4.26670017  7.0459054  10.75851601
 15.74067864 22.86313174]}} $
  
El polinomio  9 es: 
$ - \frac{x^{9}}{362880} + \frac{x^{8}}{4480} - \frac{x^{7}}{140} + \frac{7 x^{6}}{60} - \frac{21 x^{5}}{20} + \frac{21 x^{4}}{4} - 14 x^{3} + 18 x^{2} - 9 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.15232223  0.80722002  2.00513516  3.78347397  6.20495678  9.37298525
 13.46623691 18.83359779 26.37407189]}} $
  
El polinomio  10 es: 
$ \frac{x^{10}}{3628800} - \frac{x^{9}}{36288} + \frac{x^{8}}{896} - \frac{x^{7}}{42} + \frac{7 x^{6}}{24} - \frac{21 x^{5}}{10} + \frac{35 x^{4}}{4} - 20 x^{3} + \frac{45 x^{2}}{2} - 10 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.13779347  0.72945455  1.8083429   3.4014337   5.55249614  8.33015275
 11.84378584 16.27925783 21.99658581 29.92069701]}} $
  
El polinomio  11 es: 
$ - \frac{x^{11}}{39916800} + \frac{11 x^{10}}{3628800} - \frac{11 x^{9}}{72576} + \frac{11 x^{8}}{2688} - \frac{11 x^{7}}{168} + \frac{77 x^{6}}{120} - \frac{77 x^{5}}{20} + \frac{55 x^{4}}{4} - \frac{55 x^{3}}{2} + \frac{55 x^{2}}{2} - 11 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.12579644  0.66541826  1.64715055  3.09113814  5.0292844   7.50988786
 10.605951   14.43161376 19.1788574  25.21770934 33.49719285]}} $
  
El polinomio  12 es: 
$ \frac{x^{12}}{479001600} - \frac{x^{11}}{3326400} + \frac{11 x^{10}}{604800} - \frac{11 x^{9}}{18144} + \frac{11 x^{8}}{896} - \frac{11 x^{7}}{70} + \frac{77 x^{6}}{60} - \frac{33 x^{5}}{5} + \frac{165 x^{4}}{8} - \frac{110 x^{3}}{3} + 33 x^{2} - 12 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.11572212  0.61175748  1.51261027  2.83375134  4.59922764  6.84452545
  9.62131684 13.00605499 17.11685519 22.15109038 28.48796725 37.09912104]}} $
  
El polinomio  13 es: 
$ - \frac{x^{13}}{6227020800} + \frac{13 x^{12}}{479001600} - \frac{13 x^{11}}{6652800} + \frac{143 x^{10}}{1814400} - \frac{143 x^{9}}{72576} + \frac{143 x^{8}}{4480} - \frac{143 x^{7}}{420} + \frac{143 x^{6}}{60} - \frac{429 x^{5}}{40} + \frac{715 x^{4}}{24} - \frac{143 x^{3}}{3} + 39 x^{2} - 13 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.10714239  0.5661319   1.39856434  2.61659711  4.23884593  6.29225627
  8.81500194 11.86140359 15.51076204 19.88463566 25.18526386 40.72300867]}} $
  
El polinomio  14 es: 
$ \frac{x^{14}}{87178291200} - \frac{x^{13}}{444787200} + \frac{13 x^{12}}{68428800} - \frac{13 x^{11}}{1425600} + \frac{143 x^{10}}{518400} - \frac{143 x^{9}}{25920} + \frac{143 x^{8}}{1920} - \frac{143 x^{7}}{210} + \frac{1001 x^{6}}{240} - \frac{1001 x^{5}}{60} + \frac{1001 x^{4}}{24} - \frac{182 x^{3}}{3} + \frac{91 x^{2}}{2} - 14 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.09974751  0.52685765  1.30062912  2.43080108  3.93210282  5.82553622
  8.14024014 10.91649951 14.21080501 18.10489222 22.72338163 28.27298172
 35.14944366 44.36608171]}} $
  
El polinomio  15 es: 
$ - \frac{x^{15}}{1307674368000} + \frac{x^{14}}{5811886080} - \frac{x^{13}}{59304960} + \frac{13 x^{12}}{13685760} - \frac{13 x^{11}}{380160} + \frac{143 x^{10}}{172800} - \frac{143 x^{9}}{10368} + \frac{143 x^{8}}{896} - \frac{143 x^{7}}{112} + \frac{1001 x^{6}}{144} - \frac{1001 x^{5}}{40} + \frac{455 x^{4}}{8} - \frac{455 x^{3}}{6} + \frac{105 x^{2}}{2} - 15 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.09330781  0.49269174  1.21559541  2.26994953  3.66762272  5.42533663
  7.56591623 10.12022857 13.13028248 16.65440771 20.7764789  31.40751917
 38.53068331 48.02608557]}} $
  
El polinomio  16 es: 
$ \frac{x^{16}}{20922789888000} - \frac{x^{15}}{81729648000} + \frac{x^{14}}{726485760} - \frac{x^{13}}{11119680} + \frac{13 x^{12}}{3421440} - \frac{13 x^{11}}{118800} + \frac{143 x^{10}}{64800} - \frac{143 x^{9}}{4536} + \frac{143 x^{8}}{448} - \frac{143 x^{7}}{63} + \frac{1001 x^{6}}{90} - \frac{182 x^{5}}{5} + \frac{455 x^{4}}{6} - \frac{280 x^{3}}{3} + 60 x^{2} - 16 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.08764941  0.46269633  1.14105777  2.12928365  3.43708663  5.07801861
  7.07033854  9.43831434 12.21422337 15.44152737 19.18015685 19.18015686
 23.5159057  28.57872974 51.70116034]}} $
  
El polinomio  17 es: 
$ - \frac{x^{17}}{355687428096000} + \frac{17 x^{16}}{20922789888000} - \frac{17 x^{15}}{163459296000} + \frac{17 x^{14}}{2179457280} - \frac{17 x^{13}}{44478720} + \frac{221 x^{12}}{17107200} - \frac{221 x^{11}}{712800} + \frac{2431 x^{10}}{453600} - \frac{2431 x^{9}}{36288} + \frac{2431 x^{8}}{4032} - \frac{2431 x^{7}}{630} + \frac{1547 x^{6}}{90} - \frac{1547 x^{5}}{30} + \frac{595 x^{4}}{6} - \frac{340 x^{3}}{3} + 68 x^{2} - 17 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.08263821  0.43615032  1.07517658  2.00519353  3.23425612  4.77351351
  6.63782921  8.84668551 11.42552932 14.40782304 17.83828473 21.77826826
 31.58177168 45.37571653 55.38975179]}} $
  
El polinomio  18 es: 
$ \frac{x^{18}}{6402373705728000} - \frac{x^{17}}{19760412672000} + \frac{17 x^{16}}{2324754432000} - \frac{17 x^{15}}{27243216000} + \frac{17 x^{14}}{484323840} - \frac{17 x^{13}}{12355200} + \frac{221 x^{12}}{5702400} - \frac{221 x^{11}}{277200} + \frac{2431 x^{10}}{201600} - \frac{2431 x^{9}}{18144} + \frac{2431 x^{8}}{2240} - \frac{221 x^{7}}{35} + \frac{1547 x^{6}}{60} - \frac{357 x^{5}}{5} + \frac{255 x^{4}}{2} - 136 x^{3} + \frac{153 x^{2}}{2} - 18 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.07816917  0.41249009  1.01652018  1.89488851  3.05435311  4.50420554
  6.25672507  8.32782516 10.73799005 13.51365621 16.68930627 20.31076764
 29.16820868 34.62792705 41.04181678 59.09054644]}} $
  
El polinomio  19 es: 
$ - \frac{x^{19}}{121645100408832000} + \frac{19 x^{18}}{6402373705728000} - \frac{19 x^{17}}{39520825344000} + \frac{323 x^{16}}{6974263296000} - \frac{323 x^{15}}{108972864000} + \frac{323 x^{14}}{2421619200} - \frac{323 x^{13}}{74131200} + \frac{4199 x^{12}}{39916800} - \frac{4199 x^{11}}{2217600} + \frac{46189 x^{10}}{1814400} - \frac{46189 x^{9}}{181440} + \frac{4199 x^{8}}{2240} - \frac{4199 x^{7}}{420} + \frac{2261 x^{6}}{60} - \frac{969 x^{5}}{10} + \frac{323 x^{4}}{2} - \frac{323 x^{3}}{2} + \frac{171 x^{2}}{2} - 19 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.07415878  0.39126861  0.96395734  1.79617558  2.89365138  4.26421554
  5.91814156  7.86861891  7.86861892 10.13242372 12.73088146 15.69127835
 19.04899319 19.04899322 22.85084978 27.16066928 27.1606693  32.06912223
 32.06912225 32.06912228 44.31736281 62.80242315]}} $
  
El polinomio  20 es: 
$ \frac{x^{20}}{2432902008176640000} - \frac{x^{19}}{6082255020441600} + \frac{19 x^{18}}{640237370572800} - \frac{19 x^{17}}{5928123801600} + \frac{323 x^{16}}{1394852659200} - \frac{323 x^{15}}{27243216000} + \frac{323 x^{14}}{726485760} - \frac{323 x^{13}}{25945920} + \frac{4199 x^{12}}{15966720} - \frac{4199 x^{11}}{997920} + \frac{46189 x^{10}}{907200} - \frac{4199 x^{9}}{9072} + \frac{4199 x^{8}}{1344} - \frac{323 x^{7}}{21} + \frac{323 x^{6}}{6} - \frac{646 x^{5}}{5} + \frac{1615 x^{4}}{8} - 190 x^{3} + 95 x^{2} - 20 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.07053989  0.37212682  0.9165821   1.70730653  2.74919926  4.04892531
  5.61517497  7.45901745  9.59439287 12.03880253 12.03880255 14.81429346
 14.81429348 21.47878825 29.93255469 29.93255473 35.01343418 40.83305708
 47.61999403 66.52441653]}} $
  
El polinomio  21 es: 
$ - \frac{x^{21}}{51090942171709440000} + \frac{x^{20}}{115852476579840000} - \frac{x^{19}}{579262382899200} + \frac{19 x^{18}}{91462481510400} - \frac{19 x^{17}}{1129166438400} + \frac{323 x^{16}}{332107776000} - \frac{323 x^{15}}{7783776000} + \frac{323 x^{14}}{242161920} - \frac{323 x^{13}}{9884160} + \frac{4199 x^{12}}{6842880} - \frac{4199 x^{11}}{475200} + \frac{4199 x^{10}}{43200} - \frac{4199 x^{9}}{5184} + \frac{323 x^{8}}{64} - \frac{323 x^{7}}{14} + \frac{2261 x^{6}}{30} - \frac{6783 x^{5}}{40} + \frac{1995 x^{4}}{8} - \frac{665 x^{3}}{3} + 105 x^{2} - 21 x + 1 $
Sus raices son: 
$ \mathtt{\text{[6.72578200e-02 3.54772900e-01 8.73660170e-01 1.62686994e+00
 2.61862641e+00 3.85465214e+00 5.34236928e+00 7.09116881e+00
 9.11277885e+00 1.14217717e+01 1.14217718e+01 1.69789526e+01
 2.02785094e+01 2.02785096e+01 2.39718457e+01 2.81075280e+01
 5.93259941e+01 7.02556886e+01]}} $
  
El polinomio  22 es: 
$ \frac{x^{22}}{1124000727777607680000} - \frac{x^{21}}{2322315553259520000} + \frac{x^{20}}{10532043325440000} - \frac{x^{19}}{78990324940800} + \frac{19 x^{18}}{16629542092800} - \frac{19 x^{17}}{256628736000} + \frac{323 x^{16}}{90574848000} - \frac{323 x^{15}}{2476656000} + \frac{323 x^{14}}{88058880} - \frac{323 x^{13}}{4043520} + \frac{4199 x^{12}}{3110400} - \frac{4199 x^{11}}{237600} + \frac{46189 x^{10}}{259200} - \frac{3553 x^{9}}{2592} + \frac{3553 x^{8}}{448} - \frac{3553 x^{7}}{105} + \frac{24871 x^{6}}{240} - \frac{4389 x^{5}}{20} + \frac{7315 x^{4}}{24} - \frac{770 x^{3}}{3} + \frac{231 x^{2}}{2} - 22 x + 1 $
Sus raices son: 
$ \mathtt{\text{[6.42676300e-02 3.38967250e-01 8.34589990e-01 1.55371339e+00
 2.50000624e+00 3.67842034e+00 5.09534907e+00 6.75883552e+00
 8.67885331e+00 1.08676869e+01 1.08676869e+01 1.08676869e+01
 1.33404510e+01 1.33404510e+01 1.92170035e+01 1.92170036e+01
 1.92170036e+01 2.26733169e+01 2.26733179e+01 3.08133590e+01
 3.56133355e+01 4.10169682e+01 6.28570962e+01 7.39955070e+01]}} $
  
El polinomio  23 es: 
$ - \frac{x^{23}}{25852016738884976640000} + \frac{23 x^{22}}{1124000727777607680000} - \frac{23 x^{21}}{4644631106519040000} + \frac{23 x^{20}}{31596129976320000} - \frac{23 x^{19}}{315961299763200} + \frac{437 x^{18}}{83147710464000} - \frac{437 x^{17}}{1539772416000} + \frac{7429 x^{16}}{634023936000} - \frac{7429 x^{15}}{19813248000} + \frac{7429 x^{14}}{792529920} - \frac{7429 x^{13}}{40435200} + \frac{96577 x^{12}}{34214400} - \frac{96577 x^{11}}{2851200} + \frac{81719 x^{10}}{259200} - \frac{81719 x^{9}}{36288} + \frac{81719 x^{8}}{6720} - \frac{81719 x^{7}}{1680} + \frac{33649 x^{6}}{240} - \frac{33649 x^{5}}{120} + \frac{8855 x^{4}}{24} - \frac{1771 x^{3}}{6} + \frac{253 x^{2}}{2} - 23 x + 1 $
Sus raices son: 
$ \mathtt{\text{[6.15320400e-02 3.24511330e-01 7.98873970e-01 1.48688633e+00
 2.39175583e+00 3.51779744e+00 4.87056037e+00 6.45699131e+00
 8.28565189e+00 8.28565190e+00 1.03670090e+01 1.03670090e+01
 1.03670090e+01 1.03670090e+01 1.03670090e+01 1.27138251e+01
 1.27138252e+01 1.53416870e+01 1.82697427e+01 2.15217257e+01
 4.40698075e+01 4.40698124e+01 7.77432273e+01]}} $
  
El polinomio  24 es: 
$ \frac{x^{24}}{620448401733239439360000} - \frac{x^{23}}{1077167364120207360000} + \frac{23 x^{22}}{93666727314800640000} - \frac{23 x^{21}}{580578888314880000} + \frac{23 x^{20}}{5266021662720000} - \frac{23 x^{19}}{65825270784000} + \frac{437 x^{18}}{20786927616000} - \frac{437 x^{17}}{449100288000} + \frac{7429 x^{16}}{211341312000} - \frac{7429 x^{15}}{7429968000} + \frac{7429 x^{14}}{330220800} - \frac{7429 x^{13}}{18532800} + \frac{96577 x^{12}}{17107200} - \frac{7429 x^{11}}{118800} + \frac{81719 x^{10}}{151200} - \frac{81719 x^{9}}{22680} + \frac{81719 x^{8}}{4480} - \frac{4807 x^{7}}{70} + \frac{33649 x^{6}}{180} - \frac{1771 x^{5}}{5} + \frac{1771 x^{4}}{4} - \frac{1012 x^{3}}{3} + 138 x^{2} - 24 x + 1 $
Sus raices son: 
$ \mathtt{\text{[ 0.05901985  0.31123915  0.76609691  1.42559759  2.29256206  3.37077426
  4.6650837   6.18153512  7.92753925  9.91209797  9.91209803  9.91209805
 12.1461026  12.1461028  14.64273183 14.6427321  20.49146179 27.63593661
 47.15310619]}} $
  
El polinomio  25 es: 
$ - \frac{x^{25}}{15511210043330985984000000} + \frac{x^{24}}{24817936069329577574400} - \frac{x^{23}}{86173389129616588800} + \frac{23 x^{22}}{11240007277776076800} - \frac{23 x^{21}}{92892622130380800} + \frac{23 x^{20}}{1053204332544000} - \frac{23 x^{19}}{15798064988160} + \frac{437 x^{18}}{5820339732480} - \frac{437 x^{17}}{143712092160} + \frac{7429 x^{16}}{76082872320} - \frac{7429 x^{15}}{2971987200} + \frac{7429 x^{14}}{145297152} - \frac{7429 x^{13}}{8895744} + \frac{7429 x^{12}}{684288} - \frac{7429 x^{11}}{66528} + \frac{81719 x^{10}}{90720} - \frac{408595 x^{9}}{72576} + \frac{24035 x^{8}}{896} - \frac{24035 x^{7}}{252} + \frac{8855 x^{6}}{36} - \frac{1771 x^{5}}{4} + \frac{6325 x^{4}}{12} - \frac{1150 x^{3}}{3} + 150 x^{2} - 25 x + 1 $
Sus raices son: 
$ \mathtt{\text{[5.67047700e-02 5.67047800e-02 2.99010900e-01 7.35909560e-01
 1.36918312e+00 2.20132605e+00 3.23567580e+00 4.47649661e+00
 4.47649662e+00 5.92908376e+00 7.59989929e+00 7.59989931e+00
 7.59989932e+00 9.49674920e+00 9.49674924e+00 9.49674925e+00
 9.49674926e+00 1.16290149e+01 1.16290149e+01 1.16290150e+01
 1.16290150e+01 1.40079584e+01 1.66471243e+01 1.66471251e+01
 1.66471263e+01 1.95628935e+01 2.27752300e+01 2.63087593e+01
 3.91906095e+01 3.91906191e+01 8.52601556e+01 8.52601556e+01]}} $
  
El polinomio  26 es: 
$ \frac{x^{26}}{403291461126605635584000000} - \frac{x^{25}}{596585001666576384000000} + \frac{x^{24}}{1909072005333044428800} - \frac{x^{23}}{9943083361109606400} + \frac{23 x^{22}}{1729231888888627200} - \frac{23 x^{21}}{17863965794304000} + \frac{23 x^{20}}{243047153664000} - \frac{23 x^{19}}{4253325189120} + \frac{437 x^{18}}{1790873763840} - \frac{437 x^{17}}{49746493440} + \frac{7429 x^{16}}{29262643200} - \frac{7429 x^{15}}{1257379200} + \frac{7429 x^{14}}{67060224} - \frac{7429 x^{13}}{4447872} + \frac{96577 x^{12}}{4790016} - \frac{96577 x^{11}}{498960} + \frac{1062347 x^{10}}{725760} - \frac{312455 x^{9}}{36288} + \frac{312455 x^{8}}{8064} - \frac{16445 x^{7}}{126} + \frac{23023 x^{6}}{72} - \frac{3289 x^{5}}{6} + \frac{7475 x^{4}}{12} - \frac{1300 x^{3}}{3} + \frac{325 x^{2}}{2} - 26 x + 1 $
Sus raices son: 
$ \mathtt{\text{[-5.28339942e+00  5.45644800e-02  2.87707980e-01  7.08016020e-01
  1.31708137e+00  2.11712094e+00  3.11109388e+00  4.30277088e+00
  5.69681882e+00  5.69681883e+00  7.29890995e+00  7.29890996e+00
  7.29890997e+00  7.29890998e+00  9.11586283e+00  9.11586287e+00
  1.11558248e+01  1.11558249e+01  1.11558249e+01  1.11558250e+01
  1.34285081e+01  1.34285086e+01  1.34285094e+01  1.34285108e+01
  1.59455024e+01  1.87206854e+01  2.51160522e+01  3.27986037e+01
  3.72071068e+01  4.20586018e+01  8.90284027e+01]}} $
  
El polinomio  27 es: 
$ - \frac{x^{27}}{10888869450418352160768000000} + \frac{x^{26}}{14936720782466875392000000} - \frac{x^{25}}{44191481604931584000000} + \frac{x^{24}}{212119111703671603200} - \frac{x^{23}}{1473049386831052800} + \frac{23 x^{22}}{320228127571968000} - \frac{23 x^{21}}{3969770176512000} + \frac{23 x^{20}}{63012225024000} - \frac{23 x^{19}}{1260244500480} + \frac{437 x^{18}}{596957921280} - \frac{437 x^{17}}{18424627200} + \frac{7429 x^{16}}{11921817600} - \frac{7429 x^{15}}{558835200} + \frac{7429 x^{14}}{32288256} - \frac{7429 x^{13}}{2306304} + \frac{96577 x^{12}}{2661120} - \frac{96577 x^{11}}{295680} + \frac{62491 x^{10}}{26880} - \frac{312455 x^{9}}{24192} + \frac{49335 x^{8}}{896} - \frac{9867 x^{7}}{56} + \frac{3289 x^{6}}{8} - \frac{2691 x^{5}}{4} + \frac{2925 x^{4}}{4} - \frac{975 x^{3}}{2} + \frac{351 x^{2}}{2} - 27 x + 1 $
Sus raices son: 
$ \mathtt{\text{[5.25799000e-02 2.77229110e-01 6.82163910e-01 1.26881418e+00
 2.03915928e+00 2.99583558e+00 4.14219448e+00 5.48237171e+00
 5.48237172e+00 7.02137602e+00 7.02137603e+00 7.02137604e+00
 8.76520261e+00 8.76520271e+00 8.76520276e+00 1.07209789e+01
 1.07209791e+01 1.07209793e+01 1.28971502e+01 1.28971516e+01
 1.28971530e+01 1.53037228e+01 1.53037299e+01 1.79525552e+01
 2.40366673e+01 3.13024129e+01 3.99806202e+01 4.49599231e+01
 4.49636308e+01 8.07127635e+01 9.28026135e+01 9.28026136e+01
 9.28026136e+01]}} $
  
El polinomio  28 es: 
$ \frac{x^{28}}{304888344611713860501504000000} - \frac{x^{27}}{388888194657798291456000000} + \frac{x^{26}}{1066908627319062528000000} - \frac{x^{25}}{4734801600528384000000} + \frac{x^{24}}{30302730243381657600} - \frac{x^{23}}{263044533362688000} + \frac{23 x^{22}}{68620313051136000} - \frac{23 x^{21}}{992442544128000} + \frac{23 x^{20}}{18003492864000} - \frac{23 x^{19}}{405078589440} + \frac{437 x^{18}}{213199257600} - \frac{437 x^{17}}{7238246400} + \frac{7429 x^{16}}{5109350400} - \frac{7429 x^{15}}{259459200} + \frac{7429 x^{14}}{16144128} - \frac{7429 x^{13}}{1235520} + \frac{96577 x^{12}}{1520640} - \frac{5681 x^{11}}{10560} + \frac{62491 x^{10}}{17280} - \frac{16445 x^{9}}{864} + \frac{9867 x^{8}}{128} - \frac{3289 x^{7}}{14} + \frac{2093 x^{6}}{4} - 819 x^{5} + \frac{6825 x^{4}}{8} - 546 x^{3} + 189 x^{2} - 28 x + 1 $
Sus raices son: 
$ \mathtt{\text{[-5.94951730e-01  5.07346200e-02  2.67487270e-01  6.58136630e-01
  1.22397181e+00  1.96676761e+00  2.88888333e+00  3.99331166e+00
  5.28373606e+00  6.76460339e+00  6.76460340e+00  6.76460341e+00
  6.76460342e+00  8.44121625e+00  8.44121626e+00  1.03198508e+01
  1.03198509e+01  1.03198510e+01  1.24079003e+01  1.24079028e+01
  1.24079030e+01  1.24079042e+01  2.00238501e+01  2.30535977e+01
  2.30539397e+01  3.83670817e+01  8.43178359e+01  9.65824207e+01]}} $
  
El polinomio  29 es: 
$ - \frac{x^{29}}{8841761993739701954543616000000} + \frac{29 x^{28}}{304888344611713860501504000000} - \frac{29 x^{27}}{777776389315596582912000000} + \frac{29 x^{26}}{3200725881957187584000000} - \frac{29 x^{25}}{18939206402113536000000} + \frac{29 x^{24}}{151513651216908288000} - \frac{29 x^{23}}{1578267200176128000} + \frac{667 x^{22}}{480342191357952000} - \frac{667 x^{21}}{7939540353024000} + \frac{667 x^{20}}{162031435776000} - \frac{667 x^{19}}{4050785894400} + \frac{12673 x^{18}}{2345191833600} - \frac{12673 x^{17}}{86858956800} + \frac{215441 x^{16}}{66421555200} - \frac{215441 x^{15}}{3632428800} + \frac{215441 x^{14}}{242161920} - \frac{215441 x^{13}}{19768320} + \frac{164749 x^{12}}{1520640} - \frac{164749 x^{11}}{190080} + \frac{95381 x^{10}}{17280} - \frac{95381 x^{9}}{3456} + \frac{95381 x^{8}}{896} - \frac{8671 x^{7}}{28} + \frac{2639 x^{6}}{4} - \frac{7917 x^{5}}{8} + \frac{7917 x^{4}}{8} - 609 x^{3} + 203 x^{2} - 29 x + 1 $
Sus raices son: 
$ \mathtt{\text{[-8.45329366e+01  4.90144900e-02  2.58407300e-01  6.35747220e-01
  1.18220106e+00  1.89936650e+00  2.78936354e+00  3.85487616e+00
  5.09920009e+00  6.52630286e+00  6.52630287e+00  6.52630288e+00
  6.52630289e+00  6.52630290e+00  8.14089960e+00  8.14089975e+00
  8.14089989e+00  9.94854872e+00  1.19557712e+01  1.19557722e+01
  1.19557750e+01  1.41702014e+01  1.41702109e+01  2.53043524e+01
  2.87238453e+01  4.56291963e+01  1.00367492e+02  1.00367492e+02]}} $
  